spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -1,11 +1,22 @@
 RSpec.describe "expect { ... }.to raise_error" do
   it_behaves_like("an RSpec matcher", :valid_value => lambda { raise "boom" },
                                       :invalid_value => lambda { }) do
-    let(:matcher) { raise_error }
+    let(:matcher) { raise_error Exception }
   end
 
   it "passes if anything is raised" do
-    expect {raise}.to raise_error
+    expect { raise "error" }.to raise_error "error"
+  end
+
+  it "issues a warning when used without an error class or message" do
+    expect_warning_with_call_site __FILE__, __LINE__+1, /without providing a specific error/
+    expect { raise }.to raise_error
+  end
+
+  it "can supresses the warning when configured to do so" do
+    RSpec::Matchers::BuiltIn::RaiseError.allow_bare_raise_error = true
+    expect { raise }.to raise_error
+    RSpec::Matchers::BuiltIn::RaiseError.allow_bare_raise_error = false
```

Comment:
Is it worth having some examples specifically for the cases that do not issue a warning?  e.g.:

``` ruby
it 'does not issue a warning when an exception class is specified (even if it is just `Exception`)'
it 'does not issue a warning when a message is specified'
it 'does not issue a warning when a block is passed'
```